frontend/src/pages/NewAgreement.tsx

```diff
@@ -111,22 +136,97 @@ const NewAgreement: FC = () => {
   /**
    * Rationale
    */
-  const [rationale, setRationale] = useState(''); //@todo default to value in extant record if one exists
-  const [isRationaleEdited, setIsRationaleEdited] = useState(false); //@todo set true when name is loaded from extant record in Hasura
+  const [rationale, setRationale] = useState('');
+  const [isRationaleEdited, setIsRationaleEdited] = useState(false);
 
   /**
-   * "Continue" and insert new agreement
+   * Agreement insert
    */
   const [
     addAgreement,
     { data: addAgreementData, loading: addAgreementLoading, error: addAgreementError },
-  ] = useMutation(ADD_AGREEMENT, { refetchQueries: ['agreements'] });
+  ] = useMutation(addAgreementMutation, {
+    refetchQueries: ['agreements'],
+    onCompleted: (data) => {
+      // When new agreement is saved and new record id comes back from hasura, update state and sessionStorage
+      if (data?.insert_core_agreements_one.id && agreementId === null) {
+        const newAgreementId = data.insert_core_agreements_one.id;
+        sessionStorage.setItem(sessionStorageKey, newAgreementId as unknown as string);
+        setAgreementId(newAgreementId);
+        setSavedData(data.insert_core_agreements_one);
+      }
+    },
+  });
+  if (addAgreementError) console.log(addAgreementError);
+
+  /**
+   * Agreement update
+   */
+  const [
+    updateAgreement,
+    {
+      // data: updateAgreementData,
+      // loading: updateAgreementLoading,
+      error: updateAgreementError,
+    },
+  ] = useMutation(updateAgreementMutation, {
+    refetchQueries: ['agreements'],
+    onCompleted: (data) => data && setSavedData(data.update_core_agreements_by_pk),
+  });
+  if (updateAgreementError) console.log(updateAgreementError);
+
```

Comment:
We shouldn't update an agreement during its creation flow. We should only create it after everything is in place.
Consider the following scenario - a user begins creating an agreement, inserts name, rationale, topics and sections, and then closes the browser window without ever submitting the form.

Reply:
@avirankatz We just discussed this together with Yishai on the demo call. Without all data persisted to Hasura during step 2, either there are multiple sources of truth or the user risks losing all their work, neither of which is good. We just need to decide how/when exactly to clean up abandoned data. If we add a `core_agreements.published` (or inversely `core_agreements.draft`) status then we can simply decide on some mechanism to remove abandoned agreements based on this status and `updated_at`. In step 3, the status would be updated and the agreement_id removed from sessionStorage.

Reply:
Ok we can do that as well. One thing - instead of separating drafts from published agreements by using a table, let's do so by using a column such as status. That way we won't have to maintain two separate tables (if more fields are added to an agreement, for example, we would only need to update the schema of a single table).

Reply:
Yes, I think we are proposing the same thing: to add a `core_agreements.published` column. Or `core_agreements.status` if you foresee statuses beyond just _published_ and _draft_.